File: llm_generated_code/Gemini/CPP/Level2/Q2.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int minOperations(vector<int>& a) {
    int n = a.size();
    int evenCount = 0, oddCount = 0;

    for (int num : a) {
        if (num % 2 == 0) {
            evenCount++;
        } else {
            oddCount++;
        }
    }

    int targetParity = evenCount > oddCount ? 0 : 1; // 0 for even, 1 for odd

    int operations = 0;
    while (true) {
        int minOppositeParity = INT_MAX;
        int minTargetParity = INT_MAX;
        int oppositeParityIndex = -1, targetParityIndex = -1;

        for (int i = 0; i < n; i++) {
            if (a[i] % 2 != targetParity) {
                if (a[i] < minOppositeParity) {
                    minOppositeParity = a[i];
                    oppositeParityIndex = i;
                }
            } else {
                if (a[i] < minTargetParity) {
                    minTargetParity = a[i];
                    targetParityIndex = i;
                }
            }
        }

        if (oppositeParityIndex == -1 || targetParityIndex == -1) {
            break;
        }

        a[oppositeParityIndex] += a[targetParityIndex];
        operations++;
    }

    return operations;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;

        vector<int> a(n);
        for (int i = 0; i < n; i++) {
            cin >> a[i];
        }

        int result = minOperations(a);
        cout << result << endl;
    }

    return 0;
}

File: llm_generated_code/Gemini/Python/Level1/Q2.py
def min_operations(a):
    """
    Calculates the minimum number of operations to make all elements of the array have the same parity.

    Args:
        a: A list of positive integers.

    Returns:
        The minimum number of operations.
    """

    even_count = sum(1 for num in a if num % 2 == 0)
    odd_count = len(a) - even_count

    target_parity = 0 if even_count > odd_count else 1

    operations = 0
    while True:
        min_opposite_parity = float('inf')
        min_target_parity = float('inf')
        opposite_parity_index = -1
        target_parity_index = -1

        for i, num in enumerate(a):
            if num % 2 != target_parity:
                if num < min_opposite_parity:
                    min_opposite_parity = num
                    opposite_parity_index = i
            else:
                if num < min_target_parity:
                    min_target_parity = num
                    target_parity_index = i

        if opposite_parity_index == -1 or target_parity_index == -1:
            break

        a[opposite_parity_index] += a[target_parity_index]
        operations += 1

    return operations

if __name__ == "__main__":
    t = int(input())

    for _ in range(t):
        n = int(input())
        a = list(map(int, input().split()))

        result = min_operations(a)
        print(result)
